### ISSUE TITLE
Accessibility (Hard) - Keyboard (Guideline 2.1.1)

### ISSUE THREAD

**Issue by bryantal** [BCStudentSoftwareDevTeam/lsf#143]
**[Original report](https://bitbucket.org/laborstudents/lsf-flask/issue/143) by Hailey Barnett (Bitbucket: [barnetth](https://bitbucket.org/barnetth), GitHub: [barnetth](https://github.com/barnetth)).**

----------------------------------------

All functionality should be accessible through the keyboard. I have listed the places where I have found keyboard accessibility errors.

* The “My Students” and “My Department” links on the sidebar under “Supervisor Portal” are not accessible. I can tab into them, but when I click ENTER nothing happens.
* On the Labor History page for specific students, I cannot tab into the link for the specific labor position. I also cannot tab into the return button.
* The calendars throughout the interface are only accessible through mouse. I cannot tab into them to change the dates, and I cannot type out a date to change the date. This is a huge error.
* There is a problem with accessing icons through keyboards. I would imagine this is due to the icons not have the appropriate ARIA labels \(see issue #142\), but I am not positive. Icons include the question marks that appear on the Labor Status Form page, the notes icon, the remove icons, etc. **It is important to note that the pencil icon on the “Pending Forms” page does work. I would recommend comparing that icon with the other icons.**
* If you need more information about this guideline, please visit the wiki \([http://172.31.2.178/accessibility/checklist](http://172.31.2.178/accessibility/checklist)\) or ask Hailey.

![](https://bitbucket.org/repo/ey5grgE/images/4145720915-SupervisorPortalLinks_LI.jpg)
‌

![](https://bitbucket.org/repo/ey5grgE/images/3818100483-LaborHistory_LI.jpg)
‌

![](https://bitbucket.org/repo/ey5grgE/images/111083548-Calendar_LI.jpg)
‌

![](https://bitbucket.org/repo/ey5grgE/images/98784212-QuestionMarks_LI.jpg)
‌

![](https://bitbucket.org/repo/ey5grgE/images/1687784246-Icons_LI.jpg)
‌

**Comment by bryantal** [comment]
**Original comment by Hailey Barnett (Bitbucket: [barnetth](https://bitbucket.org/barnetth), GitHub: [barnetth](https://github.com/barnetth)).**

----------------------------------------

Potential calendar fix? [https://www.deque.com/blog/accessible-jquery-ui-datepicker/](https://www.deque.com/blog/accessible-jquery-ui-datepicker/)

**Comment by bryantal** [comment]
**Original comment by Hila Manalai (Bitbucket: [Hila Manalai](https://bitbucket.org/Hila Manalai), ).**

----------------------------------------

The first, second, and fourth issues are resolved. Working on the date picker accessibility.

**Comment by bryantal** [comment]
**Original comment by Hila Manalai (Bitbucket: [Hila Manalai](https://bitbucket.org/Hila Manalai), ).**

----------------------------------------

The date pickers are actually keyboard accessible. They have their own keyboard shortcuts on how to navigate through them. The keyboard shortcuts can be found in this page → [https://api.jqueryui.com/datepicker/](https://api.jqueryui.com/datepicker/)

**Comment by bryantal** [comment]
**Original comment by Audrey Craft (Bitbucket: [Audrey Craft](https://bitbucket.org/Audrey Craft), ).**

----------------------------------------

PR is in. Waiting to be merged.
